pkg/compactor/bucket-index-block-ids-fetcher.go

```diff
@@ -0,0 +1,65 @@
+package compactor
```

Comment:
Can we move this file under pkg/storage/tsdb somewhere?
I think this fetcher might be used by other components in the future as well so better to put there